back/app/services/project_copy_service.rb

```diff
@@ -63,19 +63,21 @@ def export(
     end
 
     if include_ideas
-      @template['models']['user']                = yml_users anonymize_users, shift_timestamps: shift_timestamps
-      @template['models']['basket']              = yml_baskets shift_timestamps: shift_timestamps
-      @template['models']['idea']                = yml_ideas shift_timestamps: shift_timestamps
-      @template['models']['baskets_idea']        = yml_baskets_ideas shift_timestamps: shift_timestamps
-      @template['models']['idea_file']           = yml_idea_files shift_timestamps: shift_timestamps
-      @template['models']['idea_image']          = yml_idea_images shift_timestamps: shift_timestamps
-      @template['models']['ideas_phase']         = yml_ideas_phases shift_timestamps: shift_timestamps
-      @template['models']['comment']             = yml_comments shift_timestamps: shift_timestamps
-      @template['models']['official_feedback']   = yml_official_feedback shift_timestamps: shift_timestamps
-      @template['models']['reaction'] = yml_reactions shift_timestamps: shift_timestamps
-      @template['models']['follower'] = yml_followers shift_timestamps: shift_timestamps
+      exported_ideas = @project.ideas.published
```

Comment:
Why not use an instance var, rather than passing this into the various methods?

Reply:
It's a more functional way of implementing things, which I think suits the project copy service better. It doesn't seem natural for the list of published ideas (or the project, for that matter) to be part of the service's state. While it can be convenient at times, there's always the risk of state leaking between calls. It's always a trade-off, but for this service, I don't think using an instance variable is necessary or beneficial.